File: HW2/invest_running_tests.py
import invest

portfolio=invest.Portfolio()
print(portfolio)
portfolio.history()

portfolio.addCash(300.50)
print(portfolio)
portfolio.history()

portfolio.withdrawCash(50)
portfolio.history()
print(portfolio)

s = invest.Stock(20 , "HFH")
s.name
s.price 
portfolio.buyStock(5, s) 
portfolio.history() 
print(portfolio) 

mf1 = invest.MutualFund(5, "BRT")
mf1.name
mf1.price

mf2 = invest.MutualFund (2, "GHT")
mf2.price
mf2.name

portfolio.buyMutualFund(10.3 , mf1)
print(portfolio)
portfolio.history()

portfolio.buyMutualFund (2, mf2)
print(portfolio)
portfolio.history()

portfolio.sellMutualFund("BRT", 3)
portfolio.history()
print(portfolio)

portfolio.sellStock("HFH", 1)
portfolio.history()
print(portfolio)

# check math at each stage
# then run real tests

File: HW2/invest_running_tests.rb
require_relative "invest"

port = Portfolio.new()
port.print

port.addCash(300.50)
port.print


s = Stock.new(20, "HFH")
s.print

port.buyStock(5, s)
port.print

mf1 = MutualFund.new("BRT")
mf1.print
mf2 = MutualFund.new("GHT")
mf2.print
port.buyMutualFund(10.3, mf1)
port.print
port.buyMutualFund(2, mf2)
port.print

port.sellStock("HFH", 1)
port.print

port.sellMutualFund("BRT", 3)
port.print

port.withdrawCash(50)
port.print

puts "History of this portfolio..."
port.history
